### PR TITLE
GH Action - check app prop

### PR DESCRIPTION
**Pull request by andrewjschuang**
Verifies that the component has the corresponding app prop on PR submission.

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4693"></a>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
   },
   "private": true,
   "dependencies": {
+    "@actions/core": "^1.10.0",
```
Comment (tjk) [inline package.json:65]: Is there a reason we use npm AND pnpm? (see both `package-lock.json` and `pnpm-lock.yaml`)

File: scripts/checkComponentAppProp.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const key = component.key;
+
+  const props = Object.values(component.props)
+    .filter((prop) => prop.type === "app");
+
+  if (props.length === 0) {
+    console.error(`[!] ${key} - missing app prop`);
+    err = true;
+  } else if (props.length > 1) {
+    console.error(`[!] ${key} - cannot import more than one app prop`);
```
Comment (tjk) [inline scripts/checkComponentAppProp.js:17]: nitpick: "more than one app prop not expected"

File: scripts/checkComponentAppProp.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    err = true;
+  } else {
+    const appProp = props.reduce((prop) => prop);
```
Comment (tjk) [inline scripts/checkComponentAppProp.js:20]: interesting approach than to just `props[0]` :)

**Review by tjk — DISMISSED** [review]
non-blocking comments, assuming these workflows work as expected strategy for now looks great

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[GH_actions_check_app_prop.pdf](https://github.com/PipedreamHQ/pipedream/files/9971981/GH_actions_check_app_prop.pdf)

**Review by luancazarine — APPROVED** [review]

**Pull request merged**